integration/query_array_test.go

```diff
@@ -137,3 +137,84 @@ func TestQueryArraySize(t *testing.T) {
 		})
 	}
 }
+
+// TestQueryArrayAll covers the case where the $all operator is used on an array or scalar.
+func TestQueryArrayAll(t *testing.T) {
+	t.Parallel()
+	ctx, collection := setup(t)
+
+	// Valid cases when the value for $all is an array.
+	_, err := collection.InsertMany(ctx, []any{
```

Comment:
Any reason we don't use shared data?

Reply:
Well, I just followed the approach used for `$size`. What do you mean by shared data? Should I make data between `$size` and `$all` shared, or should I reuse some other data?

Reply:
See `integration/shareddata` package and other tests that use it. `$size` test was made before we had shared arrays.

Reply:
Ok, I had a look. Those examples wouldn't fit. I need a few combinations of arrays where `$all` could and couldn't find values of various types - from one array, from a few arrays etc. The examples from shared data don't contain such cases. I don't want to extend shared data because it might affect the other cases where shared data is used, and those tests will need refactoring too.

Overall, shared data looks like an antipattern to me. I believe data needed for a test should be within the scope of that test. Why? Because of that exact reason - as soon as a new test wants to extend shared data to have more cases, the other tests will need to be refactored too to reuse those cases. 

Probably, it's fine to use individual cases from shared data (like we do with fixtures when writing tests), but using the whole data set doesn't look reasonable to me. If it's unclear, let's discuss.

Reply:
Let's add some new data to `shareddata.Composite` and use `shareddata.Scalar`, `shareddata.Scalar` and some additional test-specific data